test/functional/p2p_invalid_messages.py

```diff
@@ -42,13 +48,33 @@ def set_test_params(self):
         self.setup_clean_chain = True
 
     def run_test(self):
+        self.test_buffer()
         self.test_magic_bytes()
         self.test_checksum()
         self.test_size()
         self.test_msgtype()
         self.test_large_inv()
         self.test_resource_exhaustion()
 
+    def test_buffer(self):
+        self.log.info("Test message with header split across two buffers, should be received")
+        conn = self.nodes[0].add_p2p_connection(P2PDataStore())
+        # Create valid message
+        msg = conn.build_message(msg_ping(nonce=12345))
+        cut_pos = 12    # Chosen at an arbitrary position within the header
```

Comment:
nit: I think the fact that `cut_pos` < smallest message size is pretty smart and non-arbitrary :)
```suggestion
        cut_pos = 12    # Smaller than any other P2P message and thus easily distinguishable
```

Reply:
:+1: